odoo/addons/base/models/ir_mail_server.py

```diff
@@ -556,6 +591,80 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.model
+    def _find_mail_server(self, mail_from):
+        """Find the appropriate mail server for the given email address.
+
+        Returns: Record<ir.mail_server>, mail_from
+        - Mail server to use to send the email (None if we use the odoo-bin arguments)
+        - Email FROM to use to send the email (in some case, it might be impossible
+          to use the given email address directly if no mail server is configured for)
+        """
+        normalized_mail_from = email_normalize(mail_from)
```

Comment:
Would use prefixed variables, like: email_from, email_from_normalized, email_from_domain, ... that way it is easier to understand when reading code, you know to which original parameter the variable refers to.